tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/OpenTelemetrySpan.java

```diff
@@ -89,6 +96,33 @@ public Scope activate() {
         return new OpenTelemetryScope(delegate.makeCurrent());
     }
 
+    @Override
+    public Span baggage(String key, String value) {
+        Objects.requireNonNull(key, "Baggage Key cannot be null");
+        Objects.requireNonNull(value, "Baggage Value cannot be null");
+
+        Baggage.builder()
+                .put(key, value)
+                .build()
+                .storeInContext(getContext())
+                .makeCurrent();
+        return (io.helidon.tracing.Span) delegate;
+    }
+
+    @Override
+    public Optional<String> baggage(String key) {
+        Objects.requireNonNull(key, "Baggage Key cannot be null");
+        return Optional.ofNullable(Baggage.fromContext(getContext()).getEntryValue(key));
+    }
+
+    // Check if OTEL Context is already available in Global Helidon Context.
+    // If not – use Current context.
+    private static Context getContext() {
+        return Contexts.globalContext().get(Context.class)
```

Comment:
The same problem was already reported - do not use global context. Also do not cast to context, as it is already correctly typed.
This is the correct code:
```
return Contexts.context()
                .flatMap(ctx -> ctx.get(Context.class))
                .orElseGet(Context::current);
```

Reply:
It's NOT fixed! Please don't mark conversations as resolved if it's not fixed!